packages/babel-preset-expo/src/client-module-proxy-plugin.ts

```diff
@@ -24,50 +24,118 @@ export function reactClientReferencesPlugin(): babel.PluginObj {
+            proxyModule.push(`export default ${getProxy(exportName)};`);
+          } else {
+            proxyModule.push(`export const ${exportName} = ${getProxy(exportName)};`);
+          }
+        };
+
+        // Collect all of the exports
+        path.traverse({
+          ExportNamedDeclaration(exportPath) {
+            if (exportPath.node.declaration) {
+              if (exportPath.node.declaration.type === 'VariableDeclaration') {
+                exportPath.node.declaration.declarations.forEach((declaration) => {
+                  if (declaration.id.type === 'Identifier') {
+                    const exportName = declaration.id.name;
+                    pushProxy(exportName);
                   }
+                });
+              } else if (exportPath.node.declaration.type === 'FunctionDeclaration') {
+                const exportName = exportPath.node.declaration.id?.name;
+                if (exportName) {
+                  pushProxy(exportName);
+                }
+              } else if (exportPath.node.declaration.type === 'ClassDeclaration') {
+                const exportName = exportPath.node.declaration.id?.name;
+                if (exportName) {
+                  pushProxy(exportName);
                 }
+              } else if (
+                !['InterfaceDeclaration', 'TypeAlias'].includes(exportPath.node.declaration.type)
+              ) {
+                // TODO: What is this type?
+                console.warn(
+                  '[babel-preset-expo] Unsupported export specifier for "use client":',
+                  exportPath.node.declaration.type
+                );
               }
-            })()`
-          );
+            } else {
+              exportPath.node.specifiers.forEach((specifier) => {
+                if (types.isIdentifier(specifier.exported)) {
+                  const exportName = specifier.exported.name;
+                  pushProxy(exportName);
+                } else {
+                  // TODO: What is this type?
+                  console.warn(
+                    '[babel-preset-expo] Unsupported export specifier for "use client":',
+                    specifier
+                  );
+                }
+              });
+            }
+          },
+          ExportDefaultDeclaration() {
+            pushProxy('default');
+          },
+        });
+
+        // Clear the body
+        path.node.body = [];
+        path.node.directives = [];
+
+        path.pushContainer('body', template.ast(proxyModule.join('\n')));
+
+        assertExpoMetadata(state.file.metadata);
+
+        // Save the client reference in the metadata.
+        if (!state.file.metadata.clientReferences) {
+          state.file.metadata.clientReferences = [];
         }
+        state.file.metadata.clientReferences.push(outputKey);
+
+        // Store the proxy export names for testing purposes.
+        state.file.metadata.proxyExports = Array.from(proxyExports);
```

Comment:
Does the same thing, I lean towards using the syntax and using Array.from only when needing to pass it in a function to a higher-order function
```suggestion
        state.file.metadata.proxyExports = [...proxyExports];
```